### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by valkyrienyanko**
### Refactor
This PR is designed to clean up `index.js`.
- Events are in their own separate folder `src/events`
- Created Logger and Loader modules `src/modules`
- Put `refreshActivity()` and `resetDailyStreak()` functions in `modules/functions.js` (The idea is to create module files for things you think you can modulate and for everything else, you can scrunch into `modules/functions.js.`)
- We use axios in `index.js` to check if Discord is up and running before connecting to keep us sane

### Side Notes
I'm not sure where you defined db in `index.js` so I defined it myself and put a warning saying you couldn't connect to the database. I would recommend making db part of client (e.g. `client.db.xxxxx`). This way you can call it anywhere.

I left a command loader function in Loader module if you want to migrate to that, up to you.

Instead of `console.log(...)` you can migrate to `client.logger.log(...)`. See `modules/Logger.js` for more functions to use.

### CODE REVIEW

**Comment by Le-SirH** [comment]
Defining a DB probably wouldn't be necessary since requests to the database are done through Mongoose called by the [User](https://github.com/TheTypingMatch/le-cash-bot/blob/master/src/models/user.model.js) model. Other than that, all that is needed is a 
 `mongodb.connect()` and a `mongoose.connect()`

**Comment by valkyrienyanko** [comment]
If you look in `index.js` you will find `mongodb.connect()` and `mongoose.connect()`.

File: index.js [hunk 1]
```diff
@@ -1,124 +1,41 @@
...
-mongoDB.connect(db.uri, db.uriParams, err => {
-    if (err) log('error', err, client)
-    else console.log('Successfully connected to database.')
-})
-
-mongoose.connect(db.uri, db.uriParams, err => {
-    if (err) log('error', err, client)
-})
+const initDatabase = () => {
+    if (db === undefined)
```
Comment (Le-SirH) [inline index.js:18]: `db` could be defined from `./config/config` as URI parameters.

File: src/events/message.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    if (user && user.name !== message.author.username) {
+        const updatedName = { name: message.author.username }
+        User.updateOne({ discordId: userId }, updatedName, err => checkErr(err, client, () => {
+            console.log(`Updated username ${user.name} to ${message.author.username}`)
+        }))
+    }
+    
+    // Add user to message reward cooldown
+    if (user && !user.banned && !messageCooldowns.includes(userId)) reward(userId, client)
+    messageCooldowns.push(userId)
```
Comment (Le-SirH) [inline src/events/message.js:24]: `messageCooldowns` is not defined and is missing a function to clear the array.

File: src/events/message.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    // Check if the message starts with a prefix
+    if (!message.content.startsWith(prefix)) return
```
Comment (Le-SirH) [inline src/events/message.js:27]: `prefix` is not defined/required from `./config/config`.

File: src/events/message.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    
+    // Handle command arguments
+    const args = message.content.slice(prefix.length).trim().split(/ +/g)
+    const cmd = args.shift().toLowerCase()
+    const generalCmds = ['help', 'register', 'total', 'leaderboard', 'ping', 'faq']
+    
+    if (message.author.bot) return
+
+    // Check if the user has an account.
+    const user = await User.findOne({ discordId: userId })
```
Comment (Le-SirH) [inline src/events/message.js:12]: The `User` model is not defined because it is not imported.

File: src/events/message.js (hunk 2 above)
Comment (valkyrienyanko) [reply src/events/message.js]: What do you mean by "_is missing a function to clear the array_"?

Comment (Le-SirH) [reply src/events/message.js]: Originally, there was an interval that would clear the array of user ID's every minute. That way, they are rewarded if their ID is not included in the array.

Comment (Le-SirH) [reply src/events/message.js]: Located [here](https://github.com/TheTypingMatch/le-cash-bot/blob/9d0603c1e56610bc77e5095246ffb31540899892/index.js#L72).

Comment (valkyrienyanko) [reply src/events/message.js]: I moved that to `ready.js` under Events folder.

**Pull request merged**